src/odemis/gui/cont/settings.py

```diff
@@ -719,3 +723,203 @@ def __init__(self, tab_panel, tab_data):
+                                                        "label": "Trigger delay",
+                                                        "tooltip": "Change the trigger delay value to "
+                                                                   "center the image."},
+                                                  change_callback=self._onUpdateTriggerDelayMD)
+
+        entry_triggerDelay.value_ctrl.SetBackgroundColour(odemis.gui.BG_COLOUR_PANEL)
+        self.ctrl_triggerDelay = entry_triggerDelay.value_ctrl
+
+        entry_magnification = create_setting_entry(self.panel_streak, "Magnification",
+                                                   self.streak_lens.magnification,
+                                                   self.streak_lens,
+                                                   conf={"control_type": odemis.gui.CONTROL_COMBO,
+                                                         "label": "Magnification",
+                                                         "tooltip": "Change the magnification of the input"
+                                                                    "optics for the streak camera system. \n"
+                                                                    "Values < 1: De-magnifying \n"
+                                                                    "Values > 1: Magnifying"})
+
+        entry_magnification.value_ctrl.SetBackgroundColour(odemis.gui.BG_COLOUR_PANEL)
+        self.combo_magnification = entry_magnification.value_ctrl
+
+        # remove border
+        self.panel_streak.GetSizer().GetItem(0).SetBorder(0)
+        self.panel_streak.Layout()
+
+    def _onUpdateTriggerDelayMD(self, value):
+        """
+        Callback method for trigger delay ctrl GUI element.
+        Overwrites the triggerDelay value in the MD after a new value was requested via the GUI.
+        """
+        cur_timeRange = self.streak_unit.timeRange.value
+        requested_triggerDelay = self.ctrl_triggerDelay.GetValue()
+        # get a copy of  MD
+        trigger2delay_MD = self.streak_delay.getMetadata()[model.MD_TIME_RANGE_TO_DELAY]
+        trigger2delay_MD[cur_timeRange] = requested_triggerDelay
+        self.streak_delay.updateMetadata({model.MD_TIME_RANGE_TO_DELAY: trigger2delay_MD})
+        # Note: updateMetadata should here never raise an exception as the UnitFloatCtrl already
+        # catches errors regarding type and out-of-range inputs
+
+        # update txt displayed in GUI
+        self._onUpdateTriggerDelayGUI(STATUS, None)
+
+    def _onUpdateTriggerDelayGUI(self, mode, filename):
+        """
+        Updates the GUI elements regarding the new trigger delay value.
+        :parameter mode: (constant) SAVE, LOAD, ERROR or STATUS for display
+        :parameter filename: (str) filename of the loaded/saved file
+        """
+        if mode == LOAD:
+            self.panel.txt_StreakCalibFilename.Value = "%s" % filename
+
+            # update triggerDelay shown in GUI
```

Comment:
This function is about updating the GUI, so this block to update the VA doesn't seem appropriate here. I'd suggest to move it to the caller, just after the `updateMetadata()`. This way, you could even write `if mode in (LOAD, SAVE)`, as the both mode would be identical.